bindata/linuxptp/ptp-daemon.yaml

```diff
@@ -23,63 +23,106 @@ spec:
-            mountPath: /etc/metrics
-            readOnly: True
-      - name: linuxptp-daemon-container
-        securityContext:
-          privileged: true
-        image: {{.Image}}
-        command: [ "/bin/bash", "-c", "--" ]
-        args: [ "/usr/local/bin/ptp --alsologtostderr -v 10" ]
-        imagePullPolicy: IfNotPresent
-        resources:
-          requests:
-            cpu: 20m
-            memory: 50Mi
-        env:
-        - name: NODE_NAME
-          valueFrom:
-            fieldRef:
-              fieldPath: spec.nodeName
-        volumeMounts:
-        - name: config-volume
-          mountPath: /etc/linuxptp
-        - name: socket-dir
-          mountPath: /var/run
+       {{ if (eq .EnableEventPublisher true) }}
+        - name: cloud-native-event-sidecar
+          image: {{ .SideCar }}
+          args:
+            - "--metrics-addr=127.0.0.1:9091"
+            - "--store-path=/store"
+            - "--transport-host={{ .EventTransportHost }}"
+            - "--api-port=8080"
+          volumeMounts:
+            - name: pubsubstore
+              mountPath: /store
+            - name: logs-socket-dir
+              mountPath: /tmp
+          ports:
+            - name: metrics-port
+              containerPort: 9091
+          env:
+            - name: PTP_PLUGIN
+              value: "true"
+            - name: ptp_holdover_timeout
+              value: "{{ .HoldOverTimeout }}"
+            - name: ptp_maxoffset_threshold
+              value: "{{ .MaxOffsetThreshold }}"
+            - name: ptp_minoffset_threshold
+              value: "{{ .MinOffsetThreshold }}"
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+        {{ end }}
+        - name: kube-rbac-proxy
+          image: {{.KubeRbacProxy}}
+          imagePullPolicy: Always
+          args:
+            - --logtostderr
+            - --secure-listen-address=:8443
+            - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
+            - --upstream=http://127.0.0.1:9091/
+            - --tls-private-key-file=/etc/metrics/tls.key
+            - --tls-cert-file=/etc/metrics/tls.crt
+          ports:
+            - containerPort: 8443
+              name: https
+          resources:
+            requests:
+              cpu: 10m
+              memory: 20Mi
+          terminationMessagePolicy: FallbackToLogsOnError
+          volumeMounts:
+            - name: linuxptp-certs
+              mountPath: /etc/metrics
+              readOnly: True
+        - name: linuxptp-daemon-container
+          securityContext:
+            privileged: true
+          image: {{.Image}}
+          command: [ "/bin/bash", "-c", "--" ]
+          args: [ "/usr/local/bin/ptp --alsologtostderr -v 10" ]
+          imagePullPolicy: Always
```

Comment:
why there was a need to change this?

Reply:
oops! That was during testing , I will revert it back to IfNotPresent